src/qseed.cpp

```diff
@@ -0,0 +1,84 @@
+#include "pkcs11.h"
+#include "eaas.h"
+#include "hsm_adapter.h"
+#include "base64.h"
+
+#include <chrono>
+#include <iomanip>
+#include <sstream>
+#include <thread>
+
+const unsigned long DEFAULT_QSEED_SIZE = 2;
+const unsigned long DEFAULT_QSEED_PERIOD = 10;
+
+std::string getTimestamp() {
+
+    std::chrono::system_clock::time_point now = std::chrono::system_clock::now();
+    std::time_t time_t_now = std::chrono::system_clock::to_time_t(now);
+    auto millis = std::chrono::duration_cast<std::chrono::milliseconds>(now.time_since_epoch() % std::chrono::seconds(1)).count();
+
+    std::stringstream ss;
+    ss << std::put_time(gmtime(&time_t_now), "%FT%T") << '.' << std::setfill('0') << std::setw(3) << millis << 'Z';
+    return ss.str();
+
+}
+
+int main() {
+
+    // Check environment variables
+    const char* qryptToken = std::getenv("QRYPT_TOKEN");
+    if (!qryptToken) {
+        throw std::runtime_error("QRYPT_TOKEN environment variable is not set");
+    }
+    unsigned long sizeInKBs = DEFAULT_QSEED_SIZE;
+    auto qseedSizeAsStr = std::getenv("QSEED_SIZE");
+    if (qseedSizeAsStr) {
+        sizeInKBs = std::stoi(qseedSizeAsStr);
+    }
+    unsigned long periodInSecs = DEFAULT_QSEED_PERIOD;
+    auto qseedPeriodAsStr = std::getenv("QSEED_PERIOD");
+    if (qseedPeriodAsStr) {
+        periodInSecs = std::stoi(qseedPeriodAsStr);
+    }
+    const char* libraryFile = std::getenv("CRYPTOKI_LIB");
```

Comment:
want to check to see if a file actually exists at this location?

Reply:
you could check to see if the string inputs are actually numbers too

Reply:
When constructing the CryptokiAdapter, it will throw an error if the library file doesn't exist.

The standard library stoi will throw an invalid argument error if the string is not a number.